releasing/README.md

```diff
@@ -595,14 +595,32 @@ git push upstream :latest_kustomize
 git tag -a latest_kustomize
 ```
 
-### Optionally build a release locally
+### Optionally build locally
 
-[localbuild.sh]: localbuild.sh
+[local-build.sh]: local-build.sh
```

Comment:
Is it possible to do this somewhere in CI so that it gets tested with each PR?

Maybe a new make target in the `Makefile`, e.g. `make test-goreleaser-build` that gets added to the [prow-presubmit-check](https://github.com/kubernetes-sigs/kustomize/blob/4fd77b3a6e331dc8849360487372b27f86541f60/Makefile#L40)?

Reply:
This is a carbon copy of `cloudbuild.sh`, so in a way, it already is done (note my inclusion of `--debug` in both this and the main `cloudbuild.sh` file). The only core difference is just the changing of `goreleaser release` to `goreleaser build`. So, that said...

I _originally_ was considering instead moving the generation of `goreleaser` config to a dedicated helper  (out of `cloudbuilder.sh`). Then,  `cloudbuilder.sh` and `local-build.sh` initialize the same way and and the only real difference is that `cloudbuilder.sh` releases and `local-build.sh` just builds.

That then suggests the next logical step of finally refactoring:
- `cloudbuilder.sh` to `cloud-release.sh` (since it is the main releaser, running in the cloud)
- `cloudbuild-local.sh` to `cloud-release-local.sh` (since it too releases, but via a binding to Cloud Build and other cloud services)

Thoughts? 🤔

Reply:
p.s. I might also want to move to whatever nomenclature is used project wide, i.e. `camelCase`, `dash-case` or `snake_case` but I'm not quite able to pin down precisely what the project standard is, so I just went with one that was natural to me (which can cause problems with consistency, heh).

Reply:
I agree with the refactoring you've described. Please do it when you have a chance. Note that we're doing a release tomorrow at 1pm PT, so that could be an opportunity to try this out if it is ready. If not, no worries; we'll commit the debug flag and windows build restoration separately and can continue setting this up without rushing.

In general we use go-style filenames, but these aren't go files, and in those cases we're wildly inconsistent. 😅 So just try to be consistent within the releasing directory and don't sweat it too much.

Reply:
Ok, just to confirm here: If that gets done, it should be done as a separate PR. With "that" being specifically in reference to the multi-file refactor (separate dedicated config script) and the "debug flag and windows build restoration" together being within the scope of this specific PR.

That's at least my reference, anyway. Just to prevent scope creep and to limit the PR strictly to solving one specific problem at a time (simplifies things for everyone). Smaller PR's are easier to review and merge. 😄

Reply:
Ok flagging this as resolved for now (keeping the broader refactoring out of this particular ticket). Thank you for reviewing.